### DISCUSSION TITLE
ediff working tree file with a commit

### DISCUSSION THREAD

**Discussion by juhp** [magit/magit#4729] [Support] [answered]
I think this would be a really useful feature (in fact a little surprised it doesn't seem to exist, right?)

It is not uncommon that one wants to compare a working file with its earlier version, eg to rework some changes.

Currently the simplest workaround might be to ediff HEAD with the commit and have a third buffer open with the current working file, but this is not very convenient of course. Some Exchange answers recommend using magit-diff for this case, but my diff is too complex to read without ediff.

Or if anyone can enlighten me with a better way I am all ears, thanks

**Comment by juhp** [comment]
Okay I think I worked out a simpler workaround:

Just view the old version of the file with magit and then do a normal ediff against the working buffer.
Actually quite simple. I still think magit-ediff could provide this feature though.

**Comment by kyleam** [comment, marked as answer]
Have you tried `magit-ediff-compare`?  If you enter just a revision (rather than a range), it compares that revision against the working tree version.

**Comment by juhp** [comment]
Thanks @kyleam you are quite right indeed!

I guess I just find the transient map between magit-ediff and magit-ediff-* confusing.

So the solution is just to use `E r <commitref>`.

Personally I think it would be nicer to write "Compare range or commit" (rather than just "Show range") in the transient menu.
(I think I have also used it before in the past, but then seems forgot it.)
